### DISCUSSION TITLE
multiple IPAddress modbusTCP

### DISCUSSION THREAD

**Discussion by Mikail3** [eModbus/eModbus#191] [Q&A] [answered]
Hi!

I have the following questions; 

1. How do I set multiple IP addresses and afterward use them in the write/read?
2. How does the program know what the IP address is? I've seen the MB.addrequest, are the values(ip, port) in the add request function?
3. Can I utilize another variable name instead of MB.?

Thanks in advance!

**Comment by Miq1** [comment, marked as answer]
1+2: Please see the [documentation](https://emodbus.github.io/modbusclient-tcp-api#bool-settargetipaddress-host-uint16_t-port--uint32_t-timeout--uint32_t-interval)!
3: yes.

**Reply by Mikail3** [reply]
Thank you @Miq1 ! 
I did go trough the documentation but to be fair, I have no clue on how to utilize this in the code.

**Reply by Miq1** [reply]
The link in my first reply points exactly to the ``setTarget()`` call you will want to use to set the server's IP and port. All subsequent ``addRequest()`` will be sent to that target server, until you will use ``setTarget()`` again to choose another.
